src/fstests/fstests.jinja2

```diff
@@ -46,6 +48,40 @@ class Job(BaseJob):
             print('XML not found.', e)
             return 'fail'
 
+    def _parse_to_json(self, node, tree, name=None):
+        if 'result' in tree:
+            node['name'] = name
+            node['result'] = tree['result']
+        else:
+            if (not name == None): node['name'] = name
```

Comment:
```python
if name is not None:
    node['name'] = name
```
But also I think there are a few issues with this part of the code.

Input data structures should not be modified in the method, that's a Python convention.  Then I'm not sure why you're passing a `node` object to the method when it's basically creating one with `name`, `result` and `child_nodes`.  And I don't quite get the logic about passing `name` which could be `None` - is this just for changing the root node name to `fstests`?

Reply:
@gctucker my last question about this PR is if I should include these changes in this commit or should  I keep them in the commit refactoring this part of the code ahead?